src/main/java/demo/multiagent/application/agents/WeatherService.java

```diff
@@ -17,7 +17,7 @@ public class WeatherService {
   private HttpClient httpClient;
 
   public WeatherService(HttpClientProvider httpClientProvider) {
-    if (!System.getenv(WEATHER_API_KEY).isEmpty()) {
+    if (System.getenv(WEATHER_API_KEY) != null && !System.getenv(WEATHER_API_KEY).isEmpty()) {
```

Comment:
This would return null when not set causing the WeatherService to silently not work..

Reply:
Auch!